src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/DefaultRazorProjectService.cs

```diff
@@ -202,6 +202,7 @@ public override void UpdateDocument(string filePath, SourceText sourceText, int
         _projectSnapshotManagerDispatcher.AssertDispatcherThread();
 
         var textDocumentPath = FilePathNormalizer.Normalize(filePath);
+        // TODO: Needs to handle multiple projects!
         if (!_snapshotResolver.TryResolveProject(textDocumentPath, out var projectSnapshot))
```

Comment:
Logical mapping here for my brain, is a document a one to many to relationship with projects?

Reply:
Short answer no, long answer no, for now, with asterisks.

There is a one-to-one mapping between a document snapshot and a project snapshot (for now?) but there is a one-to-many relationship between a document file name and the project snapshots it can appear in. I don't honestly know what the best way to represent that is right now, but thats what the next PR in my list above is going to answer, depending on whether the majority of code needs to represnt things as one-to-one or one-to-many. Its complicated by the fact that we have a DocumentContext, which is defintily one-to-one and can be used to get things like the SyntaxTree and generated code etc., but there are also methods to get that info from a document snapshot. We also have a DocumentKey class which is project file name and document file name, which is either wrong or shouldn't be called "key".

So maybe everything that needs one-to-many will end up using DocumentSnapshot, and everything that needs one-to-one will end up using DocumentContext, or maybe DocumentSnapshot will stay one-to-one and there will be some other mechanism (like roslyns `GetDocumentInCurrentContext()`) to move from one world to another.